avocado/utils/download.py

```diff
@@ -162,7 +162,7 @@ def _verify_hash(filename):
     download_failures = 0
     hash_file = _verify_hash(dst)
 
-    while not hash_file == hash_expected:
+    while not crypto.cmp_hash(hash_file, hash_expected):
         hash_file = _verify_hash(_get_file(src, dst, permissions))
         if hash_file != hash_expected:
             log.error("It seems that dst %s is corrupted" % dst)
```

Comment:
Please take a look at https://github.com/avocado-framework/avocado/pull/1325. It's a big asset fetcher proposed change and it might address what you're trying to do here. Your review there is more than welcomed.